x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -205,6 +212,8 @@ export function App({
 
   useEffect(() => {
     onAppLeave((actions) => {
+      // Clear the session when leaving Lens
+      data.search.session.clear();
```

Comment:
We definitely should not be adding side effects to the `onAppLeave` handler because of how it's called. The `onAppLeave` function is executed **immediately**, and is **not executed when the user tries to leave**. Basically it sets up a state that says "if the user were to leave in the future, prevent them".

What this means is that I don't think we have a reliable way to make a side effect happen as the user is leaving, and I don't think we should be adding any side effects to the `onAppLeave` handlers.

Reply:
Good point, I guess we can put it in an unmount handler (e.g. useUnmount() )

Reply:
Did some investigation on this area:
* during the saving process, no problem with that, as the clear can be explicitly set
* otherwise if the user clicks on another plugin/app there's no other way to intercept it within Lens and clear the session. The session check happens before the component get unmounted leading to an error page. Note that this error page happens only on the `dev` environment, while in production the session gets cleared.

Reply:
Discussed with @flash1293 and found a simpler solution to that